pkg/appmanager/as3Manager.go

```diff
@@ -687,9 +698,533 @@ func (appMgr *Manager) checkValidAS3Endpoints(obj interface{}) (
+	tempRouteConfigDecl = adc
+
+	if unifiedDecl, ok := appMgr.getUnifiedAS3Declaration(tempAs3ConfigmapDecl, tempRouteConfigDecl); ok {
+		appMgr.postAS3Declaration(unifiedDecl)
+	}
+}
+
+func (appMgr *Manager) generateAS3RouteDeclaration() as3ADC {
+	// Create Shared as3Application object
+	sharedApp := as3Application{}
+	sharedApp["class"] = "Application"
+	sharedApp["template"] = "shared"
+
+	// Process CIS Resources to create AS3 Resources
+	appMgr.processResourcesForAS3(sharedApp)
+
+	// Process CustomProfiles
+	appMgr.processCustomProfilesForAS3(sharedApp)
+
+	// Process RouteProfiles
+	appMgr.processRouteProfilesForAS3(sharedApp)
+
+	// Process IRules
+	appMgr.processIRulesForAS3(sharedApp)
+
+	// Process DataGroup to be consumed by IRule
+	appMgr.processDataGroupForAS3(sharedApp)
+
+	// Create AS3 Tenant
+	tenant := as3Tenant{
+		"class":              "Tenant",
+		as3SharedApplication: sharedApp,
+	}
+	as3JSONDecl := as3ADC{
+		DEFAULT_PARTITION: tenant,
+	}
+	return as3JSONDecl
+}
+
+func (appMgr *Manager) processResourcesForAS3(sharedApp as3Application) {
+	for _, cfg := range appMgr.resources.GetAllResources() {
+		//Create policies
+		createPoliciesDecl(cfg, sharedApp)
+
+		//Create health monitor declaration
+		createMonitorDecl(cfg, sharedApp)
+
+		//Create pools
+		createPoolDecl(cfg, sharedApp)
+
+		//Create AS3 Service for virtual server
+		createServiceDecl(cfg, sharedApp)
+	}
+}
+
+func (appMgr *Manager) processIRulesForAS3(sharedApp as3Application) {
+	// Create irule declaration
+	for _, v := range appMgr.irulesMap {
+		iRule := &as3IRules{}
+		iRule.Class = "iRule"
+		iRule.IRule = v.Code
+		sharedApp[as3FormatedString(v.Name)] = iRule
+	}
+}
+
+func (appMgr *Manager) processDataGroupForAS3(sharedApp as3Application) {
+	for idk, idg := range appMgr.intDgMap {
+		for _, dg := range idg {
+			dgMap := &as3DataGroup{}
+			dgMap.Class = "Data_Group"
+			dgMap.KeyDataType = "string"
+
+			for _, record := range dg.Records {
+				var rec as3Record
+				rec.Key = record.Name
+				// To override default Value created for CCCL for certain DG types
+				if val, ok := getDGRecordValueForAS3(idk.Name, sharedApp); ok {
+					rec.Value = val
+				} else {
+					rec.Value = as3FormatedString(record.Data)
+				}
+				dgMap.Records = append(dgMap.Records, rec)
+			}
+			sharedApp[as3FormatedString(dg.Name)] = dgMap
+		}
+	}
+}
+
+func getDGRecordValueForAS3(dgName string, sharedApp as3Application) (string, bool) {
+	switch dgName {
+	case reencryptServerSslDgName:
+		for _, v := range sharedApp {
+			if svc, ok := v.(*as3Service); ok && svc.Class == "Service_HTTPS" {
+				if val, ok := svc.ClientTLS.(*as3ResourcePointer); ok {
+					return val.BigIP, true
+				}
+				if val, ok := svc.ClientTLS.(string); ok {
+					return strings.Join([]string{"", DEFAULT_PARTITION, as3SharedApplication, val}, "/"), true
+				}
+				log.Errorf("Unable to find serverssl for Data Group: %v\n", dgName)
+			}
+		}
+	}
+	return "", false
+}
+
```

Comment:
as a practice, we need to have comments about the functions for readability and understanding what really it does.

Reply:
This needs to be taken as part of refactoring.